File: src/main/java/io/github/onepieceapi/services/impl/CharacterServiceImpl.java
package io.github.onepieceapi.services.impl;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;

import io.github.onepieceapi.model.Character;
import io.github.onepieceapi.repositories.CharacterRepository;
import io.github.onepieceapi.services.CharacterService;
import lombok.AllArgsConstructor;

@Service
@AllArgsConstructor
public class CharacterServiceImpl implements CharacterService {
    private final CharacterRepository characterRepository;

    @Override
    public Page<Character> getAll(Pageable pageable) {
        return characterRepository.findAll(pageable);
    }
}

File: src/main/kotlin/io/github/onePieceApi/services/impl/CharacterServiceImpl.kt
package io.github.onePieceApi.services.impl

import io.github.onePieceApi.model.Character
import io.github.onePieceApi.repositories.CharacterRepository
import io.github.onePieceApi.services.CharacterService
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.stereotype.Service

@Service
class CharacterServiceImpl constructor(
    private val characterRepository: CharacterRepository
) : CharacterService {

    override fun getAll(pageable: Pageable): Page<Character> {
        return characterRepository.findAll(pageable)
    }

}
